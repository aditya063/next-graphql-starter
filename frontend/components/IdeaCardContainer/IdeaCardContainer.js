/* eslint-disable operator-linebreak */
/* eslint-disable no-underscore-dangle */

import { Query } from 'react-apollo';

import DisplayLoading from '../DisplayLoading/DisplayLoading';
import DisplayError from '../DisplayError/DisplayError';
import IdeaCard from '../IdeaCard/IdeaCard';
import { CURRENT_USER_PAGINATED_IDEAS } from '../../graphql/queries';
import { pageSize } from '../../config';
import * as sc from './IdeaCardContainer.style';

const IdeaContainer = React.memo(() => {
  const displayIdeaCards = ideas =>
    ideas.map(idea => (
      <IdeaCard key={`ideaCard${idea.node.id}`} {...idea.node} />
    ));

  const handleError = error => error;

  const handleFetchMore = (fetchMore, data) => {
    fetchMore({
      variables: { after: data.currentUserPaginatedIdeas.pageInfo.endCursor },
      updateQuery: (previousResult, { fetchMoreResult }) => {
        const moreEdges = fetchMoreResult.currentUserPaginatedIdeas.edges;
        const nextPageInfo = fetchMoreResult.currentUserPaginatedIdeas.pageInfo;

        return moreEdges.length
          ? {
            ...previousResult,
            currentUserPaginatedIdeas: {
              ...previousResult.currentUserPaginatedIdeas,
              edges: [
                ...previousResult.currentUserPaginatedIdeas.edges,
                ...moreEdges
              ],
              pageInfo: nextPageInfo
            }
          }
          : previousResult;
      }
    });
  };

  return (
    <Query
      query={CURRENT_USER_PAGINATED_IDEAS}
      variables={{ first: pageSize }}
      onError={handleError}
    >
      {({ loading, error, data, fetchMore }) => {
        if (loading) return <DisplayLoading />;
        if (error) return <DisplayError error={error} />;

        return (
          <sc.IdeaContainer>
            {data.currentUserPaginatedIdeas &&
            data.currentUserPaginatedIdeas.edges &&
            data.currentUserPaginatedIdeas.edges.length ? (
              <sc.ul>
                {displayIdeaCards(data.currentUserPaginatedIdeas.edges)}
              </sc.ul>
              ) : (
                <p>Add an Idea!</p>
              )}

            {data.currentUserPaginatedIdeas &&
              data.currentUserPaginatedIdeas.pageInfo &&
              data.currentUserPaginatedIdeas.pageInfo.hasNextPage && (
                <sc.loadMoreBtn
                  type="button"
                  onClick={() => handleFetchMore(fetchMore, data)}
                >
                  Load More
                </sc.loadMoreBtn>
            )}
          </sc.IdeaContainer>
        );
      }}
    </Query>
  );
});

export default IdeaContainer;
