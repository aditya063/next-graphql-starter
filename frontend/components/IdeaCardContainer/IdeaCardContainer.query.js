import gql from 'graphql-tag';

export const ME_IDEAS_QUERY = gql`
  query ME_IDEAS_QUERY {
    getUserIdeas {
      id
      content
    }
  }
`;

export const MOCK_ME_IDEAS_QUERY = {
  request: {
    query: ME_IDEAS_QUERY
  },
  result: {
    data: {
      getUserIdeas: [
        {
          id: '1',
          content: 'mock idea 1 content'
        },
        {
          id: '2',
          content: 'mock idea 2 content'
        }
      ]
    }
  }
};

export const MOCK_ERROR_ME_IDEAS_QUERY = {
  request: {
    query: ME_IDEAS_QUERY
  },
  error: new Error('mock error')
};
