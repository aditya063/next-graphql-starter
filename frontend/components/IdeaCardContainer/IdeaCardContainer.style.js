import styled from 'styled-components';

import { greenLinkBtn } from '../Styled/button.style';

export const IdeaContainer = styled.section`
  display: flex;
  flex-direction: column;
  margin-bottom: 2rem;
`;

export const ul = styled.ul`
  padding: 0;
`;

export const loadMoreBtn = styled(greenLinkBtn)`
  align-self: center;
  appearance: none;
  background-color: ${props => props.theme.color.green};

  &:hover {
    border-color: ${props => props.theme.color.black};
  }
`;
