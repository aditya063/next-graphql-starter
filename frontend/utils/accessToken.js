import fetch from 'isomorphic-unfetch';
// import cookie from 'cookie';
// import jwt from 'jsonwebtoken';

let accessToken = '';

export const setAccessToken = token => (accessToken = token);

export const getAccessToken = () => accessToken;

export const clearAccessToken = () => (accessToken = '');

export const fetchAccessToken = async refreshToken => {
  const url =
    process.env.NODE_ENV === 'development'
      ? process.env.DEV_REFRESH_URL
      : process.env.PROD_REFRESH_URL;

  const options = {
    method: 'GET',
    credentials: 'include',
    headers: { cookie: `rt=${refreshToken}` }
  };

  try {
    const response = await fetch(url, options);

    const data = await response.json();

    return data.accessToken;
  } catch (error) {
    if (process.env.NODE_ENV === 'development')
      console.error('fetchAccessToken error: ', error);
  }
};

// export const verifyAccessToken = accessToken => {
//   try {
//     jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET);
//   } catch (error) {
//     if (process.env.NODE_ENV === 'development')
//       console.error('fetchAccessToken error: ', error);
//   }
// };

// const checkAccessToken = async cookies => {
//   const accessToken = getAccessToken();

//   if (!accessToken) return true;

//   const parsedCookies = cookie.parse(cookies);

//   if (!parsedCookies.rt) return '';

//   // const response = await fetchAccessToken(parsedCookies.rt);

//   // const data = await response.json();

//   // setAccessToken(data.accessToken);
//   // setAccessTokenExpiry(data.exp);

//   // return data.accessToken;
// };
