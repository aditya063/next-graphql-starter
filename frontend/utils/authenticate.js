import jwt from 'jsonwebtoken';

const authenticate = cookie => {
  const secret = process.env.JWT_SECRET;
  const token = cookie.slice(6);

  try {
    jwt.verify(token, secret);

    return true;
  } catch {
    return false;
  }
};

const setCache = (cache, isAuthenticated) => {
  cache.writeData({
    data: {
      isLoggedIn: isAuthenticated
    }
  });
};

export default (cache, cookie) => {
  const isAuthenticated = cookie ? authenticate(cookie) : false;

  setCache(cache, isAuthenticated);
};
