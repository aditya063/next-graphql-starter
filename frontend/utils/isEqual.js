// compare props

export default (prevProps, nextProps) => {
  const obj1 = prevProps;
  console.log('TCL: isEqual -> obj1', obj1);
  const obj2 = nextProps;
  console.log('TCL: isEqual -> obj2', obj2);

  const diff = Object.keys(obj1).reduce((result, key) => {
    if (!Object.prototype.hasOwnProperty.call(obj2, key)) {
      result.push(key);
    } else if (_.isEqual(obj1[key], obj2[key])) {
      const resultKeyIndex = result.indexOf(key);

      result.splice(resultKeyIndex, 1);
    }

    return result;
  }, Object.keys(obj2));

  console.log('TCL: areEqual -> diff', diff);

  const areEqual = !diff;

  console.log('TCL: areEqual -> areEqual', areEqual);

  return areEqual;
};
