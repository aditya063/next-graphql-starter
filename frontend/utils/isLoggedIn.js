/* eslint-disable no-console */
import * as query from '../components/wrappers/WithUser/WithUser.query';

export const isLoggedIn = async client => {
  try {
    const res = await client.query({
      query: query.ME_QUERY
    });

    return res.data.me;
  } catch (err) {
    return null;
  }
};
