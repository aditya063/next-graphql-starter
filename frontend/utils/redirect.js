import Router from 'next/router';

export const redirect = (context, target) => {
  if (context.res) {
    context.res.writeHead(303, { Location: target });
    context.res.end();
  } else {
    Router.replace(target);
  }
};
