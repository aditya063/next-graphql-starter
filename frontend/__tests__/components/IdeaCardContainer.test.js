import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { load } from '../../utils/load';
// eslint-disable-next-line max-len
import IdeaCardContainer from '../../components/IdeaCardContainer/IdeaCardContainer';
// eslint-disable-next-line max-len
import {
  MOCK_ME_IDEAS_QUERY,
  MOCK_ERROR_ME_IDEAS_QUERY
} from '../../components/IdeaCardContainer/IdeaCardContainer.query';

describe('IdeaCardContainer', () => {
  let mockProps;
  let mockQueries;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    mockQueries = [MOCK_ME_IDEAS_QUERY];
    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <IdeaCardContainer {...mockProps} />
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('matches snapshot - loading', () => {
    const snapWrap = wrapper.find('Memo()');

    expect(snapWrap).toMatchSnapshot();
  });

  it('matches snapshot - loaded', async () => {
    await load(wrapper);

    const snapWrap = wrapper.find('IdeaCardContainerstyle__IdeaContainer');

    expect(snapWrap).toMatchSnapshot();
  });

  it('matches snapshot - error', async () => {
    wrapper = mount(
      <MockedProvider mocks={[MOCK_ERROR_ME_IDEAS_QUERY]} addTypename={false}>
        <IdeaCardContainer {...mockProps} />
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );

    await load(wrapper);

    const snapWrap = wrapper.find('DisplayErrorstyle__divError');

    expect(snapWrap).toMatchSnapshot();
  });
});
