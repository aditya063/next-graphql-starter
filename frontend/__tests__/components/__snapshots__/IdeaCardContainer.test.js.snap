// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCardContainer matches snapshot - error 1`] = `
.c0 {
  padding-left: 1rem;
  border-left: 5px solid red;
}

<DisplayErrorstyle__divError>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "DisplayErrorstyle__divError-sc-16p01e0-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "padding-left:1rem;border-left:5px solid red;",
          ],
        },
        "displayName": "DisplayErrorstyle__divError",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "DisplayErrorstyle__divError-sc-16p01e0-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</DisplayErrorstyle__divError>
`;

exports[`IdeaCardContainer matches snapshot - loaded 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin: 20px;
}

.c3 {
  background: none;
  background-image: url('/static/delete.svg');
  position: absolute;
  right: -10px;
  top: -10px;
  height: 20px;
  width: 20px;
  border: none;
  outline: none;
  cursor: pointer;
}

.c3:active {
  right: -8px;
  top: -8px;
  height: 17px;
  width: 17px;
}

.c4 {
  width: 100%;
  outline: none;
  border: black solid 1px;
  padding: 5px;
  margin: 0;
  overflow-wrap: break-word;
  word-break: break-all;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  padding: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<IdeaCardContainerstyle__IdeaContainer>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "IdeaCardContainerstyle__IdeaContainer-sc-155izcs-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "display:flex;justify-content:center;",
          ],
        },
        "displayName": "IdeaCardContainerstyle__IdeaContainer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "IdeaCardContainerstyle__IdeaContainer-sc-155izcs-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      <IdeaCardContainerstyle__ul>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "IdeaCardContainerstyle__ul-sc-155izcs-1",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "padding:0;flex-grow:1;",
                ],
              },
              "displayName": "IdeaCardContainerstyle__ul",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "IdeaCardContainerstyle__ul-sc-155izcs-1",
              "target": "ul",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <ul
            className="c1"
          >
            <IdeaCard
              content="mock content"
              id="1"
              key="ideaCard1"
            >
              <IdeaCardstyle__li>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "IdeaCardstyle__li-wiedry-0",
                        "isStatic": true,
                        "lastClassName": "c2",
                        "rules": Array [
                          "display:flex;position:relative;margin:20px;",
                        ],
                      },
                      "displayName": "IdeaCardstyle__li",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "IdeaCardstyle__li-wiedry-0",
                      "target": "li",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <li
                    className="c2"
                  >
                    <Mutation
                      errorPolicy="all"
                      mutation={
                        Object {
                          "definitions": Array [
                            Object {
                              "directives": Array [],
                              "kind": "OperationDefinition",
                              "name": Object {
                                "kind": "Name",
                                "value": "DELETE_IDEA_MUTATION",
                              },
                              "operation": "mutation",
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [
                                      Object {
                                        "kind": "Argument",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "id",
                                        },
                                        "value": Object {
                                          "kind": "Variable",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                        },
                                      },
                                    ],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "deleteIdea",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                              "variableDefinitions": Array [
                                Object {
                                  "defaultValue": undefined,
                                  "directives": Array [],
                                  "kind": "VariableDefinition",
                                  "type": Object {
                                    "kind": "NonNullType",
                                    "type": Object {
                                      "kind": "NamedType",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "ID",
                                      },
                                    },
                                  },
                                  "variable": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                  },
                                },
                              ],
                            },
                          ],
                          "kind": "Document",
                          "loc": Object {
                            "end": 90,
                            "start": 0,
                          },
                        }
                      }
                      onError={[Function]}
                      refetchQueries={
                        Array [
                          Object {
                            "query": Object {
                              "definitions": Array [
                                Object {
                                  "directives": Array [],
                                  "kind": "OperationDefinition",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "ME_IDEAS_QUERY",
                                  },
                                  "operation": "query",
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "getUserIdeas",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "id",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "content",
                                              },
                                              "selectionSet": undefined,
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                  "variableDefinitions": Array [],
                                },
                              ],
                              "kind": "Document",
                              "loc": Object {
                                "end": 78,
                                "start": 0,
                              },
                            },
                          },
                        ]
                      }
                      variables={
                        Object {
                          "id": "1",
                        }
                      }
                    >
                      <IdeaCardstyle__deleteBtn
                        onClick={[Function]}
                        type="button"
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "IdeaCardstyle__deleteBtn-wiedry-1",
                                "isStatic": true,
                                "lastClassName": "c3",
                                "rules": Array [
                                  "background:none;background-image:url('/static/delete.svg');position:absolute;right:-10px;top:-10px;height:20px;width:20px;border:none;outline:none;cursor:pointer;&:active{right:-8px;top:-8px;height:17px;width:17px;}",
                                ],
                              },
                              "displayName": "IdeaCardstyle__deleteBtn",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "IdeaCardstyle__deleteBtn-wiedry-1",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          onClick={[Function]}
                          type="button"
                        >
                          <button
                            className="c3"
                            onClick={[Function]}
                            type="button"
                          />
                        </StyledComponent>
                      </IdeaCardstyle__deleteBtn>
                    </Mutation>
                    <Mutation
                      errorPolicy="all"
                      mutation={
                        Object {
                          "definitions": Array [
                            Object {
                              "directives": Array [],
                              "kind": "OperationDefinition",
                              "name": Object {
                                "kind": "Name",
                                "value": "UPDATE_IDEA_MUTATION",
                              },
                              "operation": "mutation",
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [
                                      Object {
                                        "kind": "Argument",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "id",
                                        },
                                        "value": Object {
                                          "kind": "Variable",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                        },
                                      },
                                      Object {
                                        "kind": "Argument",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "content",
                                        },
                                        "value": Object {
                                          "kind": "Variable",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "content",
                                          },
                                        },
                                      },
                                    ],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "updateIdea",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "content",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                              "variableDefinitions": Array [
                                Object {
                                  "defaultValue": undefined,
                                  "directives": Array [],
                                  "kind": "VariableDefinition",
                                  "type": Object {
                                    "kind": "NonNullType",
                                    "type": Object {
                                      "kind": "NamedType",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "ID",
                                      },
                                    },
                                  },
                                  "variable": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                  },
                                },
                                Object {
                                  "defaultValue": undefined,
                                  "directives": Array [],
                                  "kind": "VariableDefinition",
                                  "type": Object {
                                    "kind": "NonNullType",
                                    "type": Object {
                                      "kind": "NamedType",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "String",
                                      },
                                    },
                                  },
                                  "variable": Object {
                                    "kind": "Variable",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "content",
                                    },
                                  },
                                },
                              ],
                            },
                          ],
                          "kind": "Document",
                          "loc": Object {
                            "end": 142,
                            "start": 0,
                          },
                        }
                      }
                      onError={[Function]}
                      refetchQueries={
                        Array [
                          Object {
                            "query": Object {
                              "definitions": Array [
                                Object {
                                  "directives": Array [],
                                  "kind": "OperationDefinition",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "ME_IDEAS_QUERY",
                                  },
                                  "operation": "query",
                                  "selectionSet": Object {
                                    "kind": "SelectionSet",
                                    "selections": Array [
                                      Object {
                                        "alias": undefined,
                                        "arguments": Array [],
                                        "directives": Array [],
                                        "kind": "Field",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "getUserIdeas",
                                        },
                                        "selectionSet": Object {
                                          "kind": "SelectionSet",
                                          "selections": Array [
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "id",
                                              },
                                              "selectionSet": undefined,
                                            },
                                            Object {
                                              "alias": undefined,
                                              "arguments": Array [],
                                              "directives": Array [],
                                              "kind": "Field",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "content",
                                              },
                                              "selectionSet": undefined,
                                            },
                                          ],
                                        },
                                      },
                                    ],
                                  },
                                  "variableDefinitions": Array [],
                                },
                              ],
                              "kind": "Document",
                              "loc": Object {
                                "end": 78,
                                "start": 0,
                              },
                            },
                          },
                        ]
                      }
                      variables={
                        Object {
                          "content": "mock content",
                          "id": "1",
                        }
                      }
                    >
                      <IdeaCardstyle__ideaP
                        contentEditable={true}
                        onInput={[Function]}
                        suppressContentEditableWarning={true}
                      >
                        <StyledComponent
                          contentEditable={true}
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "IdeaCardstyle__ideaP-wiedry-2",
                                "isStatic": true,
                                "lastClassName": "c4",
                                "rules": Array [
                                  "width:100%;outline:none;border:black solid 1px;padding:5px;margin:0;overflow-wrap:break-word;word-break:break-all;",
                                ],
                              },
                              "displayName": "IdeaCardstyle__ideaP",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "IdeaCardstyle__ideaP-wiedry-2",
                              "target": "p",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          onInput={[Function]}
                          suppressContentEditableWarning={true}
                        >
                          <p
                            className="c4"
                            contentEditable={true}
                            onInput={[Function]}
                            suppressContentEditableWarning={true}
                          >
                            mock content
                          </p>
                        </StyledComponent>
                      </IdeaCardstyle__ideaP>
                    </Mutation>
                  </li>
                </StyledComponent>
              </IdeaCardstyle__li>
            </IdeaCard>
          </ul>
        </StyledComponent>
      </IdeaCardContainerstyle__ul>
    </div>
  </StyledComponent>
</IdeaCardContainerstyle__IdeaContainer>
`;

exports[`IdeaCardContainer matches snapshot - loading 1`] = `
Array [
  <Memo()>
    <Query
      errorPolicy="all"
      onError={[Function]}
      query={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "ME_IDEAS_QUERY",
              },
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "getUserIdeas",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "content",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 78,
            "start": 0,
          },
        }
      }
    >
      <Memo()>
        <p>
          Loading ...
        </p>
      </Memo()>
    </Query>
  </Memo()>,
  <Memo()>
    <p>
      Loading ...
    </p>
  </Memo()>,
]
`;
